timeless/views.py

```diff
@@ -49,151 +66,144 @@ def delete(self):
-        return self.model.query.get(id)
-
-
-class DetailView(SingleObjectMixin, GenericView):
-    """
-    Example:
-        class SettingsDetailView(views.DetailView):
-            model = models.ReservationSettings
-            template_name = "restaurants/tables/create_edit.html"
-            success_url_name = "reservation_settings_list"
-            not_found_url_name = "reservation_settings_list"
-    """
-    not_found_url_name = None
-    success_url_name = None
-
-    def get(self, id=None):
+    def get_context_object_list_name(self):
         """
-        Get method fetch object from db and render it into template
+        Get context_object_list_name.
         """
-        assert self.not_found_url_name, "not_found_url_name is required"
-
-        instance = self.get_object(id)
-        if not instance:
-            return redirect(url_for(self.not_found_url_name))
-
-        return self.render_template({"instance": instance})
+        return self.context_object_list_name
 
+    def get_object_list(self):
+        """
+        Get the list of objects. If this method is not overwritten, then a
+        model variable must be declared, and it must have query.all().
+        """
+        if self.model is None:
+            raise NotImplementedError(f"{self.__class__.__name__} must define "
+                                      f"either 'model' or 'get_object_list()'")
+        return self.model.query.all()
 
-class CreateUpdateView(SingleObjectMixin, GenericView):
-    """Example
-        class SettingsCreateUpdateViewView(views.CreateUpdateView):
-            template_name = "restaurants/tables/create_edit.html"
-            success_url_name = "reservation_settings_list"
-            form = forms.SettingsForm
-    """
-    form = None
-    success_url_name = None
-    not_found_url_name = None
-
-    def get(self, id=None):
-        """Get method render create template """
-        assert self.form, "Form is required"
-        form = self.form()
-        if id:
-            instance = self.get_object(id)
-            if not instance:
-                return redirect(url_for(self.not_found_url_name))
-            form = self.form.TableForm(instance=instance)
-
-        return self.render_template({"form": form})
-
-    def post(self, id=None):
+    def get_default_context(self):
         """
-        Post method checks form validation,
-        save into db and redirect to success_url
+        Add the object list to the context.
         """
-        assert self.form, "Form is required"
-        assert self.success_url_name, "Success_url_name is required"
-        form = self.form(request.form)
-        if form.validate():
-            form.save()
-        return redirect(url_for(self.success_url_name))
+        context = super().get_default_context()
+        context[self.get_context_object_list_name()] = self.get_object_list()
+        return context
 
 
-class DeleteView(SingleObjectMixin, GenericView):
```

Comment:
@bkuzmic there was `SingleObjectMixin` which was doing one thing: took model instance by id from url.
It will be reusable in other generic views where object is needed (`DeleteView`, `UpdateView`). I think you could stay it as it was before. wdyt?

Reply:
@vryazanov I agree that it is useful. I'll leave it in and update the puzzles to use that object once generic CreateView, UpdateView and DeleteView are implemented.